### PR TITLE
loaddata: support concurrent writing to TiKV

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

a simple test shows concurrent writing improves the performance indeed. tiup playground, 1G data, 4 indexes
- LOAD DATA thread = 1: **9min 28s**
- lightning default thread = #CPU core: **6min 0s**
- LOAD DATA thread = #CPU core: **4min 52s**

however, during the lightning test my OS is very slow, I will deploy a cluster to test it later.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/importer/import.go [hunk 1]
```diff
@@ -300,8 +300,13 @@ func (e *LoadDataController) initFieldParams(plan *plannercore.LoadData) error {
 	return nil
 }
 
+var ignoreInTest = false
```
Comment (hawkingrei) [inline executor/importer/import.go:303]: I suggest that it is an atomic type to avoid data race report.

Comment (lance6716) [reply executor/importer/import.go:303]: will turn on race check in bazel in next commit

File: executor/load_data.go [hunk 2]
```diff
@@ -122,9 +110,8 @@ type commitTask struct {
 type LoadDataWorker struct {
 	UserSctx sessionctx.Context
 
-	// TODO: they should be []T to support concurrent load data.
-	encodeWorker *encodeWorker
-	commitWorker *commitWorker
+	encodeWorkers []*encodeWorker
+	commitWorkers []*commitWorker
```
Comment (D3Hunter) [inline executor/load_data.go:114]: prefer we make `encode/commit` pair into a chunk worker

Comment (lance6716) [reply executor/load_data.go:114]: I want to separate commitWorker for future use in @you06 's concurrent INSERT INTO ... SELECT FROM

**Review by D3Hunter — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: executor/importer/import.go (hunk 1 above)
Comment (lance6716) [reply executor/importer/import.go:303]: https://github.com/pingcap/tidb/pull/42667/commits/8827c5d0c2c0c2c5b4259401241cf62b56a5c9eb

**Pull request merged**